src/middleware/passport-auth.ts

```diff
@@ -0,0 +1,168 @@
+                    logger.debug('auth callback from onelogin received');
+
+                    if (userInfo.email === undefined) {
+                        logger.error('onelogin auth failed: account has no email address');
+                        cb(null, undefined, {
+                            message: 'onelogin account does not have an email, use another provider'
+                        });
+                        return;
+                    }
+
+                    try {
+                        logger.debug('checking if user has previously logged in...');
+                        const existingUser = await userRepository.findOneBy({ email: userInfo.email });
+
+                        if (existingUser && existingUser.provider !== 'onelogin') {
+                            logger.error('onelogin auth failed: email was registered via another provider');
+                            cb(null, undefined, { message: 'User is already registered via another provider' });
+                            return;
+                        }
+
+                        if (!existingUser) {
+                            logger.debug('no previous login found, creating new user');
+
+                            const user = await userRepository.save({
+                                provider: 'onelogin',
+                                providerUserId: userInfo.sub,
+                                email: userInfo.email,
+                                emailVerified: userInfo.email_verified
+                            });
+                            cb(null, user);
+                            return;
+                        }
+                        logger.debug('existing user found');
+                        cb(null, existingUser);
+                    } catch (error) {
+                        logger.error(error);
+                        cb(null, undefined, { message: 'Unknown error' });
+                    }
+                }
+            )
+        );
+    }
+
+    if (process.env.AUTH_PROVIDERS?.includes('google')) {
+        passport.use(
+            'google',
+            new GoogleStrategy(
+                {
+                    clientID: process.env.GOOGLE_CLIENT_ID || '',
+                    clientSecret: process.env.GOOGLE_CLIENT_SECRET || '',
+                    callbackURL: `${process.env.BACKEND_URL}/auth/google/callback`,
+                    scope: ['openid', 'profile', 'email']
+                },
+                async (accessToken, refreshToken, profile, cb): Promise<void> => {
+                    logger.debug('auth callback from google received');
+
+                    if (!profile?._json?.email) {
+                        logger.error('google auth failed: account has no email address');
+                        cb(null, undefined, {
+                            message: 'google account does not have an email, use another provider'
```

Comment:
For google this can actually happen if the user created an account with just a phone number (not sure if this is still a thing, but the docs say to check).

Reply:
I'm cool with doing what the Docs say...  I don't think its a thing any more and given we're limiting to those with Marvell Google logins definately won't be the case.